Sources/MlemMiddleware/Content Models/Internal/SiteVersion.swift

```diff
@@ -50,6 +50,27 @@ public enum SiteVersion: Equatable, Hashable {
     public static let v19_4: Self = .init("0.19.4")
 }
 
+public extension SiteVersion {
+    enum Feature {
+        case headerAuthentication, batchMarkRead
+        
+        var minimumVersion: SiteVersion {
+            switch self {
+            case .headerAuthentication: .v19_0
+            case .batchMarkRead: .v19_0
+            }
+        }
+    }
+    
+    /// Checks whether this SiteVersion supports the given feature. Always returns false if version unknown.
+    func suppports(_ feature: Feature) -> Bool {
+        switch self {
+        case .other: false
```

Comment:
Should we assume the feature is supported if `.other`?

```suggestion
        case .other: true
```

Reply:
`false` is the safer failure state:

| | Actually True | Actually False |
| --- | --- | --- |
| Assumed True | ✅ | Worst-case: failed  API calls |
| Assumed False | Worst-case: don't support all possible functionality | ✅ |

Reply:
That's a good point, for this particular case. Though if we plan to add cases of `Feature` such as `supportsPostHiding` etc, defaulting to `true` would be better for those cases:

| | Actually True | Actually False |
| --- | --- | --- |
| Assumed True | ✅ | Worst-case: "error" toast shown when user attempts to use feature |
| Assumed False |  Worst-case: User can't use feature at all even though Lemmy supports it  | ✅ |

Potentially we could decide whether to default to `true` or `false` via an `isNecessary: Bool` computed property of `Feature` or something. Happy to leave OOS though